### ISSUE TITLE
People keep asking about _trial_temp

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#10906]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)| @jml reported|
|-|-|
|Trac ID|trac#1328|
|Type|enhancement|
|Created|2005-11-12 08:08:06Z|
|Branch|https://github.com/twisted/twisted/tree/trial-temp-1328|

<details><summary>Searchable metadata</summary>

```
trac-id__1328 1328
type__enhancement enhancement
reporter__jml jml
priority__high high
milestone__ 
branch__branches_trial_temp_1328 branches/trial-temp-1328
branch_author__ 
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__documentation documentation
time__1131782886000000 1131782886000000
changetime__1222269640000000 1222269640000000
version__ 
owner__ 
cc__jml cc__TimothyFitz cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml commented|
|-|-|

```
#!html
<pre>
People keep asking about _trial_temp:
- What is it?
- Why doesn't Trial delete it?
- Why can't it be created in an absolute path?

Maybe it should be renamed to _trial, or _trial_files, or _trial_output?

Maybe a 'README.txt' or "whatisthis.txt" should be created in the _trial_temp
directory.

Most likely just add an entry to the faq (or the guide), because everything else
seems like too much work for what amounts to User Expectation Adjustment.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

Replying to [jml](#note_1):
> People keep asking about _trial_temp:
> - What is it?

Can a trained twisted dev give me a good one liner for this? :)

> - Why doesn't Trial delete it?

Good question..

> - Why can't it be created in an absolute path?

Is this still impossible?

> Maybe it should be renamed to _trial, or _trial_files, or _trial_output?

_trial_temp is fine with me and people are used to it by now.

> Maybe a 'README.txt' or "whatisthis.txt" should be created in the _trial_temp
> directory.

Sounds good, with that one-liner..

>
> Most likely just add an entry to the faq (or the guide), because everything else
> seems like too much work for what amounts to User Expectation Adjustment.

I suggest adding a note to the first section of the [doc/core/howto/testing.xhtml testing.xhtml](doc/core/howto/testing.xhtml testing.xhtml) page.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra commented|
|-|-|

(In [24639]) Branching to 'trial-temp-1328'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra commented|
|-|-|

raising priority cause 'people keep asking about it'.. ;)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

I'm not sure why the `<h3>` is being added here.  It doesn't seem related to the ticket.

Aside from that:

  * perhaps `<code class="shell">` for `_trial_temp`, rather than single quotes?
  * It's worth mentioning that _trial_temp is the _working directory_ for the trial process, rather than just saying tests can use it to write random data.
  * should mention `_trial_temp-<counter>` as well - these get created if two instances of trial are run in parallel from the same directory so as to avoid giving two different test runs the same temp directory.  might also be worth mentioning the lock files that go with these (if so, they're different on posix from how they are on windows)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_8"></a>|@thijstriemstra set owner to @jml|
|-|-|

Replying to [exarkun](#note_7):
> I'm not sure why the `<h3>` is being added here.  It doesn't seem related to the ticket.
>
> Aside from that:
>
>   * perhaps `<code class="shell">` for `_trial_temp`, rather than single quotes?
>   * It's worth mentioning that _trial_temp is the _working directory_ for the trial process, rather than just saying tests can use it to write random data.
>   * should mention `_trial_temp-<counter>` as well - these get created if two instances of trial are run in parallel from the same directory so as to avoid giving two different test runs the same temp directory.  

Fixed these points in r24657

> might also be worth mentioning the lock files that go with these (if so, they're different on posix from how they are on windows)

Assigning to maintainer for review, so he can provide details about the lock files.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_9"></a>|@thijstriemstra removed owner|
|-|-|

ping

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

`twisted.python.lockfile` is used to lock the `_trial_temp` directories.  On linux, this results in symlinks to pids.  On windows, directories with one file with a pid as the contents.  These will be cleaned up if trial exits normally, but if not they will be left behind.  They should be cleaned up the next time trial tries to use the directory they lock, but it's also safe to delete them manually before them if desired.

Can you work as much of that into the docs as you think is useful? :)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_11"></a>|@exarkun commented|
|-|-|

Oh, also, `<counter>` isn't a good idea in an xhtml document. ;)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_12"></a>|@thijstriemstra removed owner|
|-|-|

I fixed the points above and added a 'Trial directories' header, see r24871.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_13"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

Looks fine, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_14"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [24913]) Merge trial-temp-1328: Adding section to the testing howto describing the temporary folders and files created by Trial.

Author: thijs
Reviewer: exarkun
Fixes: [#10906](https://github.com/twisted/twisted/issues/10906)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
